src/sentry/tsdb/redissnuba.py

```diff
@@ -62,23 +62,27 @@ def dont_do_this(callargs):
 def selector_func(method, callargs):
     spec = method_specifications.get(method)
     if spec is None:
-        return ['redis']  # default backend (possibly invoke base directly instead?)
+        return 'redis'  # default backend (possibly invoke base directly instead?)
 
     operation_type, model_extractor = spec
-    backends = set([model_backends[model][operation_type] for model in model_extractor(callargs)])
+    backends = {model_backends[model][operation_type] for model in model_extractor(callargs)}
 
     assert len(backends) == 1, 'request was not directed to a single backend'
-    return list(backends)
+    return backends.pop()
```

Comment:
Made a bit more sense to just return the 1 item here instead of the caller extracting the 1 item. We're already asserting there's only 1.

Reply:
This makes a lot of sense and is a good observation, this caller has different semantics than the delegator.